lib/renderer/ipc-renderer-internal-utils.ts

```diff
@@ -4,38 +4,22 @@ import * as errorUtils from '@electron/internal/common/error-utils'
 type IPCHandler = (event: Electron.IpcRendererEvent, ...args: any[]) => any
 
 export const handle = function <T extends IPCHandler> (channel: string, handler: T) {
-  ipcRendererInternal.on(channel, (event, requestId, ...args) => {
-    new Promise(resolve => resolve(handler(event, ...args))
-    ).then(result => {
-      return [null, result]
-    }, error => {
-      return [errorUtils.serialize(error)]
-    }).then(responseArgs => {
-      event.sender.send(`${channel}_RESPONSE_${requestId}`, ...responseArgs)
-    })
+  ipcRendererInternal.on(channel, async (event, requestId, ...args) => {
+    const replyChannel = `${channel}_RESPONSE_${requestId}`
+    try {
+      event.sender.send(replyChannel, null, await handler(event, ...args))
+    } catch (error) {
+      event.sender.send(replyChannel, errorUtils.serialize(error))
+    }
   })
 }
 
-let nextId = 0
-
 export function invoke<T> (command: string, ...args: any[]) {
```

Comment:
This helper seems unnecessary, can we remove and inline the `ipcRendererInternal.invoke` calls instead

Reply:
I wanted to do it in a separate PR